sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/UnnecessaryUsingsTest.cs

```diff
@@ -38,6 +38,16 @@ public class UnnecessaryUsingsTest
                                     new CS.UnnecessaryUsings(),
                                     additionalReferences: GetAdditionalReferences());
 
+        [TestMethod]
```

Comment:
This looks like too many new files to maintain for one simple cases to me. Can you add them to existing file and demonstrate it inside namespace using?
```C#
namespace Repro_3408_Consumer
{
    using System; // Noncompliant - FP, See: https://github.com/SonarSource/sonar-dotnet/issues/3408

    public class Repro3408
    {
        public void Consumer()
        {
            var (_, y) = Repro_3408_Provider.ServiceReturningTuples.GetPair();
        }
    }
}

namespace Repro_3408_Provider
{
    using System;

    public static class ServiceReturningTuples
    {
        public static Tuple<string, int> GetPair() => Tuple.Create("a", 1);
    }
}
```

Reply:
Nice idea with usings in namespace. I've removed the files and added the code inline.

Reply:
Can you remove this test and move the fragment inside `TestCases\UnnecessaryUsings.cs` file? For simplicity and to follow our usual way of adding test cases.

Reply:
Doing this will require to change all file content to put everything what already exists in a namespace to avoid applying already existing usings. This will also require to change the code fix file accordingly.

I prefer to keep this test separated to make the problem obvious.